src/columns/fast-formula-parser.ts

```diff
@@ -0,0 +1,85 @@
+import * as glide from "../glide";
+
+import FormulaParser from 'fast-formula-parser';
+
+const run = async (formula, ...params) => {
+    var data: any[][] = [];
+
+    const parser = new FormulaParser({
+        onCell: ({_, row, col}) => {
+            return data[row - 1][col - 1];
+        },
+        onRange: (ref: { from: { row: any; col: any; }; to: { row: number; col: number; }; }) => {
+            const arr: any[][] = [];
+            for (let row = ref.from.row; row <= ref.to.row; row++) {
+                const innerArr: any[] = [];
+                if (data[row - 1]) {
+                    for (let col = ref.from.col; col <= ref.to.col; col++) {
+                        innerArr.push(data[row - 1][col - 1]);
+                    }
+                }
+                arr.push(innerArr);
+            }
+            return arr;
+        }
+    });
+
+    if (formula === undefined || formula.value === undefined) return undefined;
+    data = params.map(p => [p.value]);
```

Comment:
You can put this at the top of the function, make it a `const data = ...` and remove the `var data` there.